File: ios/SwiftKotlination/SwiftKotlinationTests/Mocks/Features/Stories/Model/StoryRepositoryMock.swift
import XCTest
import RxSwift
@testable import SwiftKotlination

final class StoryRepositoryMock: StoryRepositoryProtocol {
    var storyStub: StoryStub
    
    init(storyStub: StoryStub = .failure(.timeout)) {
        self.storyStub = storyStub
    }
    
    enum StoryStub {
        case success(Story)
        case failure(RxError)
    }
    
    var story: Observable<Story> {
        switch storyStub {
        case .success(let story):
            return Observable.just(story)
        case .failure(let error):
            return Observable.error(error)
        }
    }
    
}

File: android/SwiftKotlination/app/src/test/java/fr/jhandguy/swiftkotlination/features/story/viewmodel/StoryRepositoryMock.kt
package fr.jhandguy.swiftkotlination.features.story.viewmodel

import fr.jhandguy.swiftkotlination.features.story.model.Story
import fr.jhandguy.swiftkotlination.features.story.model.StoryRepository
import io.reactivex.Observable

class StoryRepositoryMock(override var story: Observable<Story>) : StoryRepository